execution/gethexec/sequencer.go

```diff
@@ -935,13 +952,27 @@ func (s *Sequencer) createBlock(ctx context.Context) (returnValue bool) {
 
 	start := time.Now()
 	var (
-		block *types.Block
-		err   error
+		block                      *types.Block
+		err                        error
+		shouldSequenceWithEspresso bool
 	)
+
+	// Initialize shouldSequenceWithEspresso to false and if we have a light client reader then give it a value based on hotshot liveness
+	// This is a side effect of the sequencer having the capability to run without an L1 reader. For the Espresso integration this is a necessary component of the sequencer.
+	// However, many tests use the case of having a nil l1 reader
+	if s.lightClientReader != nil {
```

Comment:
I am thinking this might cause some issues - so lets say I have `espresso` enabled but I run the sequencer without the `lightClientReader` , I will never create espresso transactions then. It could just be a honest configuration issue but I might not realize it. Let me know your thoughts on the same!

Reply:
You're right that this would be an issue if someone were to attempt to set up a sequencer node without enabling the parent chain reader, but in that case it should act as a normal nitro sequencer.
 
I think it's a fine bound to add to our sequencer that we need an l1 reader for it to function in espresso mode properly. Fundamentally we need to read from the light client to decide how to sequence with the escape hatch. Doing it this way allows normal nitro sequencers with this code base, but also adds compatibility with espresso. I think its fine to expect node operators to configure their sequencer appropriately when they want to use espresso.